biochemist-python/chapters/Processing_Multiple_Files_and_Writing_Files.ipynb

```diff
@@ -0,0 +1,427 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import glob\n",
+    "filenames = glob.glob(file_location)\n",
+    "print(filenames)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Reading multiple files with nested for loops\n",
+    "\n",
+    "Now you have a list of all the files which end in `*.pdb` in the `PDB_files` directory. To parse every file we just read in, we will use a `for` loop to go through each file."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "for f in filenames:\n",
+    "    outfile = open(f,'r')\n",
+    "    data = outfile.readlines()\n",
+    "    outfile.close()\n",
+    "    for line in data:\n",
+    "        if 'RESOLUTION.' in line:\n",
+    "            res_line = line\n",
+    "            words = res_line.split()\n",
+    "            resolution = float(words[3])\n",
+    "            print(resolution)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Notice that in this code we actually used two `for` loops, one nested inside the other.  The outer `for` loop counts over the filenames we read in earlier.  The inner `for` loop counts over the line in each file, just as we did in our previous file parsing lesson.  \n",
+    "\n",
+    "The output our code currently generates is not that useful.  It doesn't show us which file each resolution value came from.  \n",
+    "\n",
+    "We want to print the name of the molecule with the resolution. We can use `os.path.basename`, which is another function in `os.path` to get just the name of the file."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "first_file = filenames[0] # look above to recall the content of filenames\n",
+    "print(first_file)\n",
+    "\n",
+    "file_name = os.path.basename(first_file)\n",
+    "print(file_name)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "<div class=\"exercise-this-is-a-title exercise\">\n",
+    "<p class=\"exercise-title\">Check your understanding</p>\n",
+    "    <p>How would you extract the PDB ID from the example above? </p>\n",
+    "\n",
+    "\n",
+    "    \n",
+    "```{admonition} Solution\n",
+    ":class: dropdown\n",
+    "\n",
+    "You can use the str.split function introduced in the last lesson, and split at the '.' character.\n",
+    "    \n",
+    "```\n",
+    "</div>"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "split_filename = file_name.split('.')\n",
```

Comment:
This is the exercise solution. Should be repeated in expandable section.

Reply:
Done.